src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/RazorFileChangeDetectorTest.cs

```diff
@@ -74,14 +74,21 @@ public void FileSystemWatcher_RazorFileEvent_Background_NotifiesChange()
             var fileChangeDetector = new RazorFileChangeDetector(Dispatcher, FilePathNormalizer, new[] { listener.Object })
             {
                 EnqueueDelay = 50,
-                NotifyCompletedNotifications = new ManualResetEventSlim(initialState: false)
+                BlockNotificationWorkStart = new ManualResetEventSlim(initialState: false),
             };
 
             // Act
             fileChangeDetector.FileSystemWatcher_RazorFileEvent_Background(filePath, changeKind);
 
             // Assert
-            Assert.True(fileChangeDetector.NotifyCompletedNotifications.Wait(TimeSpan.FromSeconds(2)));
+
+            // We acquire the notification prior to unblockign notification work because once we allow that work to proceed the notification will be removed.
```

Comment:
```suggestion
            // We acquire the notification prior to unblocking notification work because once we allow that work to proceed the notification will be removed.
```